File: musicflow_backend/src/models/topic.model.js
const mongoose = require("mongoose");

const topicSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      unique: true,
    },
    description: {
      type: String,
      default: "",
    },
    imageUrl: {
      type: String,
      default: "",
    },
    color: {
      type: String,
      default: "#1DB954", // Màu mặc định giống Spotify
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Topic", topicSchema);

File: musicflow_app/lib/data/models/topic_model.dart
class Topic {
  final String id;
  final String name;
  final String description;
  final String avatar;

  Topic({
    required this.id,
    required this.name,
    required this.description,
    required this.avatar,
  });

  factory Topic.fromJson(Map<String, dynamic> json) {
    return Topic(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      avatar: json['avatar'] ?? '',
    );
  }
}
